supervisor_types.go

```diff
@@ -136,27 +136,34 @@ type Dimension struct {
 // SpatialDimensionSet is a unique set of druid datasource spatial dimensions.
 type SpatialDimensionSet []SpatialDimension
 
+// DimensionExclusionsSet represents set of excluded dimensions.
+type DimensionExclusionsSet []string
+
 // DimensionsSpec is responsible for configuring Druid's dimensions. They're a
 // set of columns in Druid's data model that can be used for grouping, filtering
 // or applying aggregations.
 // https://druid.apache.org/docs/latest/ingestion/ingestion-spec#dimensionsspec
 type DimensionsSpec struct {
-	Dimensions           DimensionSet        `json:"dimensions,omitempty"`
-	DimansionExclusions  DimensionSet        `json:"dimansionExclusions,omitempty"`
-	SpatialDimensions    SpatialDimensionSet `json:"spatialDimensions,omitempty"`
-	IncludeAllDimensions bool                `json:"includeAllDimensions,omitempty"`
-	UseSchemaDiscovery   bool                `json:"useSchemaDiscovery,omitempty"`
+	Dimensions           DimensionSet           `json:"dimensions,omitempty"`
+	DimensionExclusions  DimensionExclusionsSet `json:"dimensionExclusions,omitempty"`
+	SpatialDimensions    SpatialDimensionSet    `json:"spatialDimensions,omitempty"`
+	IncludeAllDimensions bool                   `json:"includeAllDimensions,omitempty"`
+	UseSchemaDiscovery   bool                   `json:"useSchemaDiscovery,omitempty"`
+}
+type QueryGranularity struct {
+	Type string `json:"type,omitempty"`
 }
 
 // GranularitySpec allows for configuring operations such as data segment
 // partitioning, truncating timestamps, time chunk segmentation or roll-up.
 // https://druid.apache.org/docs/latest/ingestion/ingestion-spec#granularityspec
 type GranularitySpec struct {
-	Type               string   `json:"type"`
-	SegmentGranularity string   `json:"segmentGranularity,omitempty"`
-	QueryGranularity   string   `json:"queryGranularity,omitempty"`
-	Rollup             bool     `json:"rollup,omitempty"`
-	Intervals          []string `json:"intervals,omitempty"`
+	Type               string `json:"type"`
+	SegmentGranularity string `json:"segmentGranularity,omitempty"`
+	// TODO: this field is problematic as depending on value druid returns string or object
```

Comment:
Can you please rephrase? `TODO:`s should be actionable and imperative.

Imagine I'd like to help and eliminate all of the TODOs. TODO should lead me to the resources that would help me to resolve it.

Reply:
that was actually my comment directed to myself, to remind me what has to be done once I pick this PR back. Though Victor did, so he treated that as just another TODO tag.

@vzaytsev1981 we need to cover both cases of QueryGranularity here:
- when it's set to none
- when it's set to some value

Druid returns different structure (`string` for value, `object` for none) - we need to somehow handle that, @zoido's below comment seems to make a good point.

Reply:
Tx! Addressed by implementing suggestion below.